src/artemis_driver.cpp

```diff
@@ -277,6 +290,120 @@ TaskCollection ArtemisDriver<GEOM>::StepTasks() {
+    if (Globals::my_rank == 0) {
+      std::cout << "\nTaking " << nsteps << " radiation substeps"
+                << " at dt=" << dtr << std::endl;
+    }
+  }
+
+  for (int step = 1; step <= nsteps; step++, trad += dtr) {
+    status = RadiationTasks().Execute();
+    if (status != TaskListStatus::complete) {
+      return status;
+    }
+  }
+  return status;
+}
+
+template <Coordinates GEOM>
+TaskCollection ArtemisDriver<GEOM>::RadiationTasks() {
+  using TQ = TaskQualifier;
+  TaskCollection tc;
+  // Return empty TaskCollection if all unsplit physics disabled
+  if (not do_moment) return tc;
+
+  using namespace ::parthenon::Update;
+  TaskID none(0);
+  const auto any = parthenon::BoundaryType::any;
+  const int num_partitions = pmesh->DefaultNumPartitions();
+
+  // Deep copy u0 into u1 for integrator logic
+  auto &init_region = tc.AddRegion(num_partitions);
+  for (int i = 0; i < num_partitions; i++) {
+    auto &tl = init_region[i];
+    auto &u0 = pmesh->mesh_data.GetOrAdd("u0", i);
+    auto &u1 = pmesh->mesh_data.GetOrAdd("u1", i);
+    tl.AddTask(none, ArtemisUtils::DeepCopyConservedData, u1.get(), u0.get());
+  }
+
+  // Now do explicit integration of unsplit physics
+  for (int stage = 1; stage <= rad_integrator->nstages; stage++) {
+    const Real time = trad;
+    const Real bdt = rad_integrator->beta[stage - 1] * dtr;
+    const Real gam0 = rad_integrator->gam0[stage - 1];
+    const Real gam1 = rad_integrator->gam1[stage - 1];
+
+    TaskRegion &tr = tc.AddRegion(num_partitions);
+    for (int i = 0; i < num_partitions; i++) {
+      auto &tl = tr[i];
+      auto &u0 = pmesh->mesh_data.GetOrAdd("u0", i);
+      auto &u1 = pmesh->mesh_data.GetOrAdd("u1", i);
+
+      // Start looking for incoming messages (including for flux correction)
+      auto start_recv = tl.AddTask(none, parthenon::StartReceiveBoundBufs<any>, u0);
+      auto start_flx_recv = tl.AddTask(none, parthenon::StartReceiveFluxCorrections, u0);
+
+      // Compute hydrodynamic fluxes
+      auto rad_flx = tl.AddTask(none, Radiation::CalculateFluxes, u0.get());
+      auto gas_flx = tl.AddTask(none, Gas::ZeroFluxes, u0.get());
+
+      // Communicate and set fluxes
+      auto send_flx =
+          tl.AddTask(rad_flx | gas_flx,
+                     parthenon::SendBoundBufs<parthenon::BoundaryType::flxcor_send>, u0);
+      auto recv_flx = tl.AddTask(start_flx_recv, parthenon::ReceiveFluxCorrections, u0);
+      auto set_flx = tl.AddTask(recv_flx, parthenon::SetFluxCorrections, u0);
+
+      // Apply flux divergence
+      auto update = tl.AddTask(rad_flx | gas_flx | set_flx, Radiation::ApplyUpdate<GEOM>,
+                               u0.get(), u1.get(), stage, gam0, gam1, bdt);
+
+      // Apply "coordinate source terms"
+      auto coord_src = tl.AddTask(update, Radiation::FluxSource, u0.get(), bdt);
+
+      // Apply matter-coupling step
+      // This is the first task to update the gas/dust values
+      auto coupling =
```

Comment:
Assuming reading this correctly, here is a 0D iterative coupling step - does this mean the radiation solver can super-cycle hydro?